### ISSUE TITLE
Storyboard compilation fails on Xcode8.0

### ISSUE THREAD

**Issue by kanersan** [facebook/infer#487]
I have installed infer version 0.9.3 which is supposed to support Xcode8.0 however when I created a single view project (nothing other than 1 viewcontroller and Main.storyboard) infer fails at the CompileStoryboard step. Attched is the build log. 

[InferBuildLog.rtf.zip](https://github.com/facebook/infer/files/551478/InferBuildLog.rtf.zip)

PS. Project is called MultipleFrameworks because I was going to add frameworks in it to test Infer with but it fails before even adding anything into the project.

**Comment by dulmarod** [comment]
Does the build command succeed on its own? i.e. xcodebuild -target MultipleFrameworks -sdk iphonesimulator

**Comment by kanersan** [comment]
Oh wow it seems that's failing too actually but the project runs fine when I run it on Xcode8.0.
The log I get when I run "xcodebuild -target MultipleFrameworks -sdk iphonesimulator": 

Build settings from command line:
    SDKROOT = iphonesimulator9.3

2016-10-26 09:41:26.903 xcodebuild[59756:5160956] CoreSimulator is attempting to unload a stale CoreSimulatorService job.  Detected Xcode.app relocation or CoreSimulatorService version change.  Framework path (/Applications/Xcode.app/Contents/Developer/Library/PrivateFrameworks/CoreSimulator.framework) and version (209.19) does not match existing job path (/Applications/Xcode8.0.app/Contents/Developer/Library/PrivateFrameworks/CoreSimulator.framework/Versions/A/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc/Contents/MacOS/com.apple.CoreSimulator.CoreSimulatorService) and version (303.8).

=== BUILD TARGET MultipleFrameworks OF PROJECT MultipleFrameworks WITH THE DEFAULT CONFIGURATION (Release) ===

Check dependencies
2016-10-26 09:41:27.157 xcodebuild[59756:5160956] Failed to locate a valid instance of CoreSimulatorService in the bootstrap.  Adding it now.
2016-10-26 09:41:27.169 xcodebuild[59756:5160956] *** Assertion failure in -[SimServiceContext reloadServiceIfMovedOrAbortIfWeAreInvalid], /BuildRoot/Library/Caches/com.apple.xbs/Sources/CoreSimulator/CoreSimulator-209.19/CoreSimulator/SimServiceContext.m:536
** INTERNAL ERROR: Uncaught exception **
Uncaught Exception: The loaded com.apple.CoreSimulator.CoreSimulatorService job does not match our expectations: pathOfLoadedJob: /Applications/Xcode8.0.app/Contents/Developer/Library/PrivateFrameworks/CoreSimulator.framework/Versions/A/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc/Contents/MacOS/com.apple.CoreSimulator.CoreSimulatorService, our frameworkPath: /Applications/Xcode.app/Contents/Developer/Library/PrivateFrameworks/CoreSimulator.framework
Stack:
  0   __exceptionPreprocess (in CoreFoundation)
  1   objc_exception_throw (in libobjc.A.dylib)
  2   +[NSException raise:format:arguments:] (in CoreFoundation)
  3   -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] (in Foundation)
  4   -[SimServiceContext reloadServiceIfMovedOrAbortIfWeAreInvalid] (in CoreSimulator)
  5   -[SimServiceContext connect] (in CoreSimulator)
  6   -[SimServiceContext initWithDeveloperDir:connectionType:] (in CoreSimulator)
  7   __63+[SimServiceContext sharedServiceContextForDeveloperDir:error:]_block_invoke_2 (in CoreSimulator)
  8   _dispatch_client_callout (in libdispatch.dylib)
  9   _dispatch_barrier_sync_f_invoke (in libdispatch.dylib)
 10   +[SimServiceContext sharedServiceContextForDeveloperDir:error:] (in CoreSimulator)
 11   __60+[SimServiceContext(DVTAdditions) dvt_sharedServiceContext:]_block_invoke (in DVTiPhoneSimulatorRemoteClient)
 12   _dispatch_client_callout (in libdispatch.dylib)
 13   dispatch_once_f (in libdispatch.dylib)
 14   +[SimServiceContext(DVTAdditions) dvt_sharedServiceContext:] (in DVTiPhoneSimulatorRemoteClient)
 15   __42-[DVTiPhoneSimulatorLocator startLocating]_block_invoke (in IDEiOSSupportCore)
 16   __DVTDispatchAsync_block_invoke (in DVTFoundation)
 17   _dispatch_call_block_and_release (in libdispatch.dylib)
 18   _dispatch_client_callout (in libdispatch.dylib)
 19   _dispatch_root_queue_drain (in libdispatch.dylib)
 20   _dispatch_worker_thread3 (in libdispatch.dylib)
 21   _pthread_wqthread (in libsystem_pthread.dylib)
 22   start_wqthread (in libsystem_pthread.dylib)

**Comment by dulmarod** [comment]
I don't know what's the correct build command. 
If you are using CocoaPods, you may need to use xcodebuild -workspace MultipleFrameworks.xcworkspace instead of target.

**Comment by kanersan** [comment]
No cocoapods in this project. There is no workspace either, its just a single view template from Xcode. That's why I'm surprised it's failing to build.

**Comment by dulmarod** [comment]
I would recommend trying to Google how to compile an ios app in the command line with xcode8 and your settings.

**Comment by kanersan** [comment]
Right, thanks for pointing me in the right direction. It wasn't my build settings or the project itself. I had Xcode 7 on my machine as well as Xcode 8. My terminal/command line was using Xcode 7 for building as that was the registered Xcode on the system. I deleted old xcode and ran this in command line: "sudo xcode-select --switch /Applications/Xcode8.0.app" Then command line xcodebuild started using Xcode8.0. Now infer actually finds no issues in the project when I ran it. 
Thank you.
